travisci/travisci.py

```diff
@@ -0,0 +1,241 @@
+            # Just in case we are exuberantly passed the entire Exception and
+            #  not its textual representation.
+            text = str(text)
+        raise BackendError(status_code=500,
+                           reason="Internal Server Error",
+                           content=text)
+
+    # pylint: disable=no-self-use
+    def raise_from_req(self, req):
+        """Turn a failed request response into a BackendError.  Handy for
+        reflecting HTTP errors from farther back in the call chain.
+        """
+        if req.status_code < 400:
+            # Request was successful.  Or at least, not a failure.
+            return
+        raise BackendError(status_code=req.status_code,
+                           reason=req.reason,
+                           content=req.text)
+
+    def exchange_token(self):
+        """Exchange a GitHub token for a TravisCI one.
+        """
+        travis_token_url = self.travis_host + "/auth/github"
+        postdata = {
+            "github_token": self.github_token
+        }
+        self._debug("Exchanging GitHub token for Travis CI token")
+        req = requests.post(travis_token_url, headers=self.travis_headers,
+                            json=postdata)
+        self.raise_from_req(req)
+        # pylint: disable=broad-except
+        try:
+            rdata = req.json()
+        except Exception as exc:
+            self.raise_ise(str(exc))
+        access_token = rdata.get("access_token")
+        if not access_token:
+            raise BackendError(status_code=403,
+                               reason="Forbidden",
+                               content="Unable to get Travis CI access token")
+        self._debug("Travis CI token acquired")
+        return access_token
+
+    def start_travis_sync(self):
+        """Initiate a Travis <-> GH Sync"""
+        sync_url = self.travis_host + "/users/sync"
+        # Start sync
+        req = requests.post(sync_url, headers=self.travis_headers)
+        if req.status_code == 409:
+            # 409 is "already syncing"; so we pretend it was ours.
+            req.status_code = 200
+        self.raise_from_req(req)
+        self._debug("Travis CI <-> GitHub Sync started")
+
+    def enable_travis_webhook(self, slug):
+        """Enable repository for Travis CI.
+        """
+        self.set_travis_webhook(slug, enabled=True)
+
+    def disable_travis_webhook(self, slug):
+        """Disable repository for Travis CI.
+        """
+        self.set_travis_webhook(slug, enabled=False)
+
+    def set_travis_webhook(self, slug, enabled=True):
+        """Enable/disable repository for Travis CI.
+        """
+        # pylint: disable=too-many-locals
+        self.start_travis_sync()
+        user_url = self.travis_host + "/repos/" + slug
+        req = self._retry_request("get", user_url,
+                                  headers=self.travis_headers)
+        # Get the ID and flip the switch
+        # pylint: disable=broad-except
```

Comment:
RWAHR. For all these req.json() ... Isn't there only a few exceptions that they could be? Or is the goal to prevent anything from bubbling up past this?